app/src/Commands/Instances/UpCommand.php

```diff
@@ -43,46 +43,51 @@ public function configure() : void
             ->setAliases(["up"])
             ->addArgument("instance", InputArgument::OPTIONAL, "Name of the instance to start")
             ->addOption("global", "g", InputOption::VALUE_NONE, "Determines if an instance is global or not")
+            ->addOption("all", "a", InputOption::VALUE_NONE, "If is set start all instances")
         ;
     }
 
     public function execute(InputInterface $input, OutputInterface $output) : int
     {
         $instance = $input->getArgument("instance");
+        $all = $input->getOption("all");
 
-        if (is_null($instance)) {
+        if (is_null($instance) && ! $all) {
             $path = $this->filesystem->getCurrentWorkingDirectory();
 
-            if (! $this->hasDockerComposeFile($path, $output)) {
-                return Command::FAILURE;
-            }
-
-            $instance = $this->filesystem->getFilenameFromPath($path);
-            if (! $this->docker->isInstanceUp($path)) {
-                $this->writer->beginBlock($output, "Start container $instance");
-                $this->docker->startContainerByDockerCompose($path);
-                $this->writer->endBlock();
-                return Command::SUCCESS;
-            }
-
-            $output->writeln("Nothing to do. Instance $instance is already up.");
-
-            return Command::SUCCESS;
+            return $this->startInstance(
+                $output,
+                $path,
+                $this->filesystem->getFilenameFromPath($path)
+            );
         }
 
         $home_dir = $this->posix->getHomeDirectory($this->posix->getUserId());
 
-        $path = "$home_dir/.doil/instances/$instance";
+        $path = "$home_dir/.doil/instances";
```

Comment:
The logic here is switched from what is used in the other commands (global / not global). 
Would be more consistent if it was the same for all commands.

Reply:
Good eyes!!!
done!